streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -53,34 +85,121 @@ public void process(final K key, final Change<V> value) {
                 internalProcessorContext.forward(key, value);
             } // else skip
         } else {
-            throw new NotImplementedException();
+            ensureBuffer();
+            buffer(key, value);
+            enforceSizeBound();
         }
     }
 
-    private boolean shouldForward(final Change<V> value) {
-        return !(value.newValue == null && suppress.suppressTombstones());
+    @SuppressWarnings("unchecked")
+    private <T> Serde<T> castSerde(final Serde<?> untyped, final String keyOrValue) {
+        try {
+            return (Serde<T>) untyped;
```

Comment:
Come to think of it, this is probably insufficient to catch the wrong serde (due to erasure). I probably need to relocate this error message to the actual call to de/serialize